src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/EnrollmentRegistrationEntry.types.js

```diff
@@ -1,5 +1,16 @@
 // @flow
+import type { RenderFoundation } from '../../../metaData/RenderFoundation';
+import type { RegistrationFormMetadata } from '../common/types';
+
 export type OwnProps = $ReadOnly<{|
+  id: string,
+  enrollmentMetadata: RegistrationFormMetadata,
   selectedScopeId: string,
-  id: string
+  saveButtonText: string,
+  fieldOptions?: Object,
+  onSave: (dataEntryId: string, itemId: string, formFoundation: RenderFoundation) => void,
+  onPostProcessErrorMessage: Function,
+  onGetUnsavedAttributeValues: Function,
+  onUpdateField: Function,
+  onStartAsyncUpdateField: Function,
```

Comment:
Here I am trying to explicitly define the properties this components needs in order to be rendered. 
In the future when someone will need to use it they can look in this signature

Reply:
Good point @JoakimSM. Yep for me the naming here is under discussion as well. 
What I am trying to do with the `OwnProps` is to have a type for properties that the user passes directly on the component when they use it.

Eg.
```
<Button type='one' size='lg'>
Text
</Button>
```
On the example above `OwnProps` would be the `type` and the `size`. However sometimes you might have the `ContainerProps` which are props that are passed on the container level. My only point is that I would want to differentiate between what I call `OwnProps` and the rest of props a component might have. This is because then flow can tell you what props you are supposed to pass in the component when you try to use it. 

Again I agree we would like to align on a name here. I stole my name from the redux world, look [here](https://react-redux.js.org/using-react-redux/connect-mapstate).
---

Now if we agree on communicating the same thing we can be creative with a name. What do you think, would you find it useful to differentiate between the props that we pass on the component and the props that the component takes from redux?

Reply:
Thanks for explaining. Our use cases are a bit different. `OwnProps` is good for what you're trying to achieve here 👍